tests/std/tests/P0896R4_ranges_alg_minmax/test.cpp

```diff
@@ -361,6 +363,64 @@ void test_gh_1893() {
     ASSERT(val == "meow");
 }
 
+class input_move_iterator {
+public:
+    using iterator_category = input_iterator_tag;
+    using iterator_concept  = input_iterator_tag;
+    using difference_type   = ptrdiff_t;
+    using value_type        = shared_ptr<int>;
+    using pointer           = shared_ptr<int>*;
+    using reference         = shared_ptr<int>&&;
+
+    input_move_iterator() = default;
+    explicit input_move_iterator(shared_ptr<int>* ptr) : m_ptr(ptr) {}
+
+    reference operator*() const {
+        return ranges::iter_move(m_ptr);
+    }
+    pointer operator->() const {
+        return m_ptr;
+    }
+
+    input_move_iterator& operator++() {
+        ++m_ptr;
+        return *this;
+    }
+    input_move_iterator operator++(int) {
+        input_move_iterator tmp = *this;
+        ++*this;
+        return tmp;
+    }
+
+    friend bool operator==(const input_move_iterator& a, const input_move_iterator& b) = default;

```

Comment:
Nitpick: We conventionally don't bother naming parameters when defaulting functions.